### PR TITLE
dev-libs/eb: fix build with gcc14

### PR DESCRIPTION
**Pull request by xz-dev**
Closes: https://bugs.gentoo.org/920641

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: dev-libs/eb/eb-4.4.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (mm1ke) [inline dev-libs/eb/eb-4.4.3.ebuild:4]:
```suggestion
EAPI=8
````

File: dev-libs/eb/eb-4.4.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		$(use_enable ipv6) \
+		$(use_enable nls) \
+		$(use_enable threads pthread) \
+		--with-pkgdocdir="${EPREFIX}"/usr/share/doc/${PF}/html \
+		--disable-static
```
Comment (mm1ke) [inline dev-libs/eb/eb-4.4.3.ebuild:40]:
```suggestion
	local myeconfargs=(
		$(use_enable ipv6)
		$(use_enable nls)
		$(use_enable threads pthread)
		--with-pkgdocdir="${EPREFIX}"/usr/share/doc/${PF}/html
	)

	econf "${myeconfargs[@]}"
```
FYI, `--disable-static` shouldn't be required, it should be passed per default.

Comment (xz-dev) [reply dev-libs/eb/eb-4.4.3.ebuild]: I'm not sure how to test it.

Comment (mm1ke) [reply dev-libs/eb/eb-4.4.3.ebuild]: You can simply call `ebuild eb-4.4.3.ebuild configure` and check if `--disable-static` is passed correctly.

For example with 4.4.1 i get (with the same changes)
```
./configure --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --datarootdir=/usr/share --disable-dependency-tracking --disable-silent-rules --disable-static --docdir=/usr/share/doc/eb-4.4.1-r1 --htmldir=/usr/share/doc/eb-4.4.1-r1/html --with-sysroot=/ --libdir=/usr/lib64 --enable-ipv6 --enable-nls --enable-pthread --with-pkgdocdir=/usr/share/doc/eb-4.4.1-r1/html
```

The `--disable-static` is in the middle. 
Please note that this only works from `EAPI=8`. `EAPI=7` still would need the `--disable-static`.

**Review by juippis — COMMENTED** [review]
Can you compile eb with `+threads` use flag? It seems broken to me:
```
readtext.c: In function 'eb_is_text_stopped':
libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY -march=native -O2 -pipe -frecord-gcc-switches -c stopcode.c  -fPIC -DPIC -o .libs/stopcode.o
/bin/sh ../libtool  --tag=CC   --mode=compile x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY      -march=native -O2 -pipe -frecord-gcc-switches  -c -o widefont.lo widefont.c
readtext.c:1563:13: error: passing argument 1 of 'eb_lock' from incompatible pointer type [-Wincompatible-pointer-types[https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types]]
 1563 |     eb_lock(book);
      |             ^~~~
      |             |
      |             EB_Book * {aka struct EB_Book_Struct *}
/bin/sh ../libtool  --tag=CC   --mode=compile x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY      -march=native -O2 -pipe -frecord-gcc-switches  -c -o word.lo word.c
In file included from readtext.c:33:
build-post.h:277:23: note: expected 'EB_Lock *' {aka 'struct EB_Lock_Struct *'} but argument is of type 'EB_Book *' {aka 'struct EB_Book_Struct *'}
  277 | void eb_lock(EB_Lock *lock);
      |              ~~~~~~~~~^~~~
libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY -march=native -O2 -pipe -frecord-gcc-switches -c setword.c  -fPIC -DPIC -o .libs/setword.o
libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY -march=native -O2 -pipe -frecord-gcc-switches -c strcasecmp.c  -fPIC -DPIC -o .libs/strcasecmp.o
libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY -march=native -O2 -pipe -frecord-gcc-switches -c subbook.c  -fPIC -DPIC -o .libs/subbook.o
make[3]: *** [Makefile:571: readtext.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
```

**Comment by xz-dev** [comment]
> Can you compile eb with `+threads` use flag? It seems broken to me:
> 
> ```
> readtext.c: In function 'eb_is_text_stopped':
> libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY -march=native -O2 -pipe -frecord-gcc-switches -c stopcode.c  -fPIC -DPIC -o .libs/stopcode.o
> /bin/sh ../libtool  --tag=CC   --mode=compile x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY      -march=native -O2 -pipe -frecord-gcc-switches  -c -o widefont.lo widefont.c
> readtext.c:1563:13: error: passing argument 1 of 'eb_lock' from incompatible pointer type [-Wincompatible-pointer-types[https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types]]
>  1563 |     eb_lock(book);
>       |             ^~~~
>       |             |
>       |             EB_Book * {aka struct EB_Book_Struct *}
> /bin/sh ../libtool  --tag=CC   --mode=compile x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY      -march=native -O2 -pipe -frecord-gcc-switches  -c -o word.lo word.c
> In file included from readtext.c:33:
> build-post.h:277:23: note: expected 'EB_Lock *' {aka 'struct EB_Lock_Struct *'} but argument is of type 'EB_Book *' {aka 'struct EB_Book_Struct *'}
>   277 | void eb_lock(EB_Lock *lock);
>       |              ~~~~~~~~~^~~~
> libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY -march=native -O2 -pipe -frecord-gcc-switches -c setword.c  -fPIC -DPIC -o .libs/setword.o
> libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY -march=native -O2 -pipe -frecord-gcc-switches -c strcasecmp.c  -fPIC -DPIC -o .libs/strcasecmp.o
> libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -DEB_BUILD_LIBRARY -march=native -O2 -pipe -frecord-gcc-switches -c subbook.c  -fPIC -DPIC -o .libs/subbook.o
> make[3]: *** [Makefile:571: readtext.lo] Error 1
> make[3]: *** Waiting for unfinished jobs....
> ```

I will try to fix it.

File: dev-libs/eb/eb-4.4.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+)
+
+src_prepare() {
+	default
+	eautoreconf
+}
+
+src_configure() {
+	local myeconfargs=(
+		$(use_enable ipv6)
```
Comment (juippis) [inline dev-libs/eb/eb-4.4.3.ebuild:37]: If possible, just always enable this.

**Review by juippis — COMMENTED** [review]
I can't compile 4.4.1-r1 even with your patch at all, using gcc-14.2.1_p20240817 but 4.4.3 works. I'd also be hesitant to touch a stable ebuild like this. So I'll cherry-pick the 4.4.3 version bump sometime today (a bit busy atm).

File: dev-libs/eb/eb-4.4.3.ebuild (hunk 3 above)
Comment (xz-dev) [reply dev-libs/eb/eb-4.4.3.ebuild:37]: If you like, you can change it. Because I don't know what `ipv6` means.

**Comment by xz-dev** [comment]
> I can't compile 4.4.1-r1 even with your patch at all, using gcc-14.2.1_p20240817 but 4.4.3 works. I'd also be hesitant to touch a stable ebuild like this. So I'll cherry-pick the 4.4.3 version bump sometime today (a bit busy atm).

I can build it as normal (but the origin src_url is dead), I switch the new src_url, can you try again?

**Pull request closed without merging**